### PR TITLE
Model#11 order model

### LINKED ISSUE #11: Create Order model.
**_Order_**

> _id IntegerField PrimaryKey
> paymentMethod ChoiceField ...
> shippingAddressId IntegerField ForeignKey
> userId IntegerField ForeignKey
> orderDate DateTime
> shippingStatus ChoiceField
> paymentStatus ChoiceField

### PR DESCRIPTION
**Pull request by ruslanliska**
Created Order Model. Added all needed fields _id IntegerField PrimaryKey +
paymentMethod ChoiceField +
shippingAddressId IntegerField ForeignKey +
userId IntegerField ForeignKey +
orderDate DateTime +
shippingStatus ChoiceField +

Also created feature for creating order only for authenticated users and modifying order for admins

Resolves #11

### CODE REVIEW

File: ProjectShop/ProjectShop/settings.py [hunk 1]
```diff
@@ -41,11 +41,13 @@
 
     # 3rd party apps
     'rest_framework',
+    "order",
```
Comment (v-lavrushko) [inline ProjectShop/ProjectShop/settings.py:44]: order should be in local apps

Comment (ruslanliska) [reply ProjectShop/ProjectShop/settings.py]: > order should be in local apps

Done

File: ProjectShop/ProjectShop/urls.py [hunk 2]
```diff
@@ -15,7 +15,9 @@
 """
 from django.contrib import admin
 from django.urls import path
+from django.urls import path, include
```
Comment (MykhailoGrynenko) [inline ProjectShop/ProjectShop/urls.py:18]: Двічі імпортується path.

File: ProjectShop/order/admin.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from django.contrib import admin
+from .models import Order
+
+
+admin.site.register(Order)
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/admin.py:5]: No newline at the end of the file.

File: ProjectShop/order/admin.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+from django.contrib import admin
+from .models import Order
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/admin.py:2]: Please, add an absolute import path: from order.models import Order.

File: ProjectShop/order/urls.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from django.urls import path, include
+from django.contrib import admin
+from order.views import *
+
+app_name = "order"
+urlpatterns = [
+    path('create/', OrderCreateView.as_view()),
+    path('all/', OrderListView.as_view()),
+    path('detail/<int:pk>/', OrderDetailView.as_view()),
+]
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/urls.py:10]: No newline at the end of the file.

File: ProjectShop/order/views.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+class OrderListView(generics.ListAPIView):
+    serializer_class = OrderListSerializer
+    queryset = Order.objects.all()
+    permission_classes = (IsAuthenticated, IsAdminUser)
+
+
+class OrderDetailView(generics.RetrieveUpdateDestroyAPIView):
+    serializer_class = OrderDetailSerializer
+    queryset = Order.objects.all()
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/views.py:19]: No newline at the end of the file.

File: ProjectShop/order/models.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+
+# Create your models here.
+class Order(models.Model):
+    order_date = models.DateTimeField(auto_now_add=True)
+    PAYMENT_METHODS = (
+        (1, "Card"),
+        (2, "Cash")
+    )
+    payment_method = models.IntegerField(verbose_name="Payment Method", choices=PAYMENT_METHODS, default=2)
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/models.py:13]: I propose to add translation:
from django.utils.translation import ugettext_lazy as _

File: ProjectShop/order/migrations/0002_order_payment_status.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+    dependencies = [
+        ('order', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='order',
+            name='payment_status',
+            field=models.IntegerField(choices=[(1, 'Pending'), (2, 'Paid')], default=1, verbose_name='Payment Method'),
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/migrations/0002_order_payment_status.py:16]: Translation.

File: ProjectShop/order/models.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    user = models.ForeignKey(User, on_delete=models.CASCADE, null=True)
+    # shippingAddress_id = models.ForeignKey("ShippingAddress", on_delete=models.CASCADE)
+    SHIPPING_STATUSES = (
+        (1, "Planning"),
+        (2, "Shipping"),
+        (3, "Complete")
+    )
+    shipping_status = models.IntegerField(verbose_name="Shipping status", choices=SHIPPING_STATUSES, default=1)
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/models.py:21]: Translation.

File: ProjectShop/order/models.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    PAYMENT_STATUSES = (
+        (1, "Pending"),
+        (2, "Paid")
+    )
+    payment_status = models.IntegerField(verbose_name="Payment Status", choices=PAYMENT_STATUSES, default=1)
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/models.py:26]: Translation.

File: ProjectShop/order/views.py [hunk 11]
```diff
@@ -0,0 +1,19 @@
+from django.shortcuts import render
+from rest_framework import generics
+from order.serializers import OrderDetailSerializer
+from order.models import Order
+from order.serializers import OrderListSerializer
+from rest_framework.permissions import IsAuthenticated, IsAdminUser
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/views.py:6]: According to PEP 8:
Imports should be grouped in the following order:
1. Standard library imports.
2. Related third-party imports.
3. Local application/library-specific imports.
You should put a blank line between each group of imports.

File: ProjectShop/order/urls.py [hunk 12]
```diff
@@ -0,0 +1,10 @@
+from django.urls import path, include
+from django.contrib import admin
+from order.views import *
+
+app_name = "order"
+urlpatterns = [
+    path('create/', OrderCreateView.as_view()),
+    path('all/', OrderListView.as_view()),
+    path('detail/<int:pk>/', OrderDetailView.as_view()),
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/urls.py:9]: I guess it would be reasonable to add a third argument - 'name'.

**Review by MykhailoGrynenko — CHANGES_REQUESTED** [review]
I'd also like to see docstrings. Import ordering should be done according to PEP 8

File: ProjectShop/ProjectShop/settings.py [hunk 13]
```diff
@@ -44,8 +44,10 @@
     'phonenumber_field',
 
     # local apps
-    'UserApp'
-]
+    'UserApp',
+     "order",
```
Comment (ProkhvatylovArtem) [inline ProjectShop/ProjectShop/settings.py:48]: Double quotes aren't necessary here.

File: ProjectShop/order/models.py (hunk 7 above)
Comment (strangerok1507) [reply ProjectShop/order/models.py:35]: I propose to use enum for choices https://docs.djangoproject.com/en/3.2/ref/models/fields/#field-choices-enum-types

File: ProjectShop/order/models.py (hunk 9 above)
Comment (MykhailoGrynenko) [reply ProjectShop/order/models.py:43]: Let me make myself clear.
By translation I mean this:
from django.utils.translation import ugettext as _
https://docs.djangoproject.com/en/3.2/topics/i18n/translation/

File: ProjectShop/order/admin.py [hunk 14]
```diff
@@ -0,0 +1,7 @@
+from django.contrib import admin
+
```
Comment (OlehHnyp) [inline ProjectShop/order/admin.py:2]: Між групами імпорту достатньо однієї пустої лінії (тут і в декількох наступних файлах).

**Review by OlehHnyp — APPROVED** [review]
Все ок працює.

File: ProjectShop/order/views.py [hunk 15]
```diff
@@ -0,0 +1,28 @@
+from django.shortcuts import render
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated, IsAdminUser
+
+from order.models import Order
+from order.serializers import OrderDetailSerializer
+from order.serializers import OrderListSerializer
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/views.py:7]: I guess,
from order.serializers import OrderDetailSerializer, OrderListSerializer
From new line we import like this:
import os
import re
import datetime

File: ProjectShop/order/migrations/0002_order_payment_status.py (hunk 8 above)
Comment (MykhailoGrynenko) [inline ProjectShop/order/migrations/0002_order_payment_status.py:16]: from django.utils.translation import gettext_lazy as _
field=models.IntegerField(choices=[(1, 'Pending'), (2, 'Paid')], default=1, verbose_name=_('Payment Method')),

**Review by ruslanliska — COMMENTED** [review]
Thank you!
Docstrings were added

File: ProjectShop/order/models.py [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+
+    payment_status: Payment status of the order.
+    (1, "Pending"), (2, "Paid").
+    type: int, default: 1
+
+    """
+    order_date = models.DateTimeField(auto_now_add=True)
+
+
+    class PaymentMethod(models.TextChoices):
```
Comment (v-lavrushko) [inline ProjectShop/order/models.py:33]: можливо винести зза межі класу, або визначити перед полями? як буде правильно?

File: ProjectShop/order/models.py [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+
+# Create your models here.
+class Order(models.Model):
+    order_date = models.DateTimeField(auto_now_add=True)
+    PAYMENT_METHODS = (
+        (1, "Card"),
+        (2, "Cash")
+    )
+    payment_method = models.IntegerField(verbose_name="Payment Method", choices=PAYMENT_METHODS, default=2)
+    user = models.ForeignKey(User, on_delete=models.CASCADE, null=True)
```
[the remark this answers is not in the document]
Comment (ruslanliska) [reply ProjectShop/order/models.py:42]: Ну ми ніби говорили що незареєстрований клієнт зможе створити замовлення, тому тут буде null якщо клієнт який незареєстрований ствоює замовлення @v-lavrushko

**Review by ProkhvatylovArtem — APPROVED** [review]

**Review by CheshirLvova — APPROVED** [review]

**Comment by ruslanliska** [comment]
@strangerok1507 need your review

**Review by CheshirLvova — APPROVED** [review]

**Review by v-lavrushko — APPROVED** [review]

File: ProjectShop/order/views.py [hunk 18]
```diff
@@ -0,0 +1,10 @@
+from rest_framework import viewsets
+
+from order.models import Order
+from order.serializers import OrderDetailSerializer
+
+class OrderViewSet(viewsets.ModelViewSet):
```
[the remark this answers is not in the document]
Comment (v-lavrushko) [reply ProjectShop/order/views.py:6]: @strangerok1507 в чем разница между DELETED и CANCELED BY USER?

**Pull request merged**